### ISSUE TITLE
Syntax and prose for p:choose (no subpipelines)

### ISSUE THREAD

**Issue by xml-project** [xproc/3.0-specification#113]
Currently the syntax of p:choose allows to have neither a p:when nor a p:otherwise. The prose reflects this by saying: "The list of alternative subpipelines consists of zero or more subpipelines guarded by an XPath expression, followed optionally by a single default subpipeline."

So it is not a static error to have a <p:choose/>. But it is of course a dynamic error:
"It is a dynamic error (err:XD0004) if no subpipeline is selected by the p:choose and no default is provided."

This might seem a bit strange and we should talk about this.

**Comment by Conal-Tuohy** [comment]
So, this is currently OK?!
```
<p:choose>
   <p:otherwise>
   ...
   </p:otherwise>
</p:choose>
```

**Comment by xml-project** [comment]
Look at the syntax of XProc 1.0. Other reading?

**Comment by xml-project** [comment]
Thesis 1: <p:choose /> should be a static error.
Prove: Neither XMLCalabash nor MorganaXProc will run <p:choose /> but fail with different static errors. <p:error /> can be used to provoke a dynamic exception.

Thesis 2: At least one <p:when /> should be mandatory.
Prove: <p:choose><p:otherwise>...</p:otherwise></p:choose> is identical to <p:group>...</p:group>

Proposed resolution:
Change syntax from p:when* to p:when+. Change prose and introduce new static error raised, when p:when is missing.

**Comment by xml-project** [comment]
From telcon 04-14:
a p:chooses is required to have at least one (p:when or p:otherwise). 
The reason for allowing <p:choose><p:otherwise>...</otherwise></p:choose> is that you can comment out all p:when (for debugging reasons) and the pipeline is still syntactically correct.

**Comment by xml-project** [comment]
Fixed with PR #124
